scripts/utilities/pyrescal.py

```diff
@@ -8,33 +8,130 @@
 import subprocess
 import struct
 
+import cellspace
+import rescal_utilities
+
+
+
+
+# list of rescal parameters that are paths to files/directories that
+# already exist
+# constructor will try to verify that they exist 
+
+
+
+
 
 class PyRescal:
     '''Runs a rescal process and gets its data.'''
 
-
-    def __init__(self, parameters, processing_options, rescal_root=None):
+    def __init__(self, parameters, experiment_directory, rescal_root=None,
+                 keep_height_maps=True, keep_ffts=True, relative_parameter_paths=True):
+        '''Set up all the file locations and lists to hold the data files.'''
+        
         self.parameters = parameters
-        self.processing_options = processing_options
 
         # try to find RESCAL_SNOW_ROOT
         if rescal_root is None:
-            self.rescal_root = os.environ['RESCAL_SNOW_ROOT']
+            self.rescal_root = os.path.expanduser(os.environ['RESCAL_SNOW_ROOT'])
+        else:
+            self.rescal_root = os.path.expanduser(rescal_root)
+        self.rescal_root = os.path.abspath(self.rescal_root)
+
+        
+        
+        # deal with experiment_directory
+        # TODO deal with directory already existing
+        if not os.path.isabs(experiment_directory):
+            self.experiment_directory = os.path.join(self.rescal_root, experiment_directory)
+        else:
+            # TODO, this could break everything if user gives bad path
+            self.experiment_directory = os.fspath(experiment_directory)
+        self.experiment_directory = os.path.abspath(self.experiment_directory)
+            
+        # set up paths to other files/directories of interest
+        self.scripts_directory = os.path.join(self.rescal_root, 'scripts')
+        self.real_data_directory = os.path.join(self.scripts_directory, 'real_data')
+        self.rescal_executable = os.path.join(self.scripts_directory, 'rescal')
```

Comment:
Consider making a helper function to check for directory existence while making path variable?